### PR TITLE
Test - Create temp entry with same key as an expired entry on the ledger it would normally get evicted

### PR DESCRIPTION
**Pull request by sisuresh**
# Description

Resolves https://github.com/stellar/stellar-core/issues/4000.

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/transactions/test/InvokeHostFunctionTests.cpp [hunk 1]
```diff
@@ -2048,6 +2048,11 @@ TEST_CASE("temp entry eviction", "[tx][soroban]")
     cfg.METADATA_OUTPUT_STREAM = metaPath;
 
     ContractStorageInvocationTest test{cfg};
+    auto const& contractKeys = test.getContractKeys();
+
+    // extend WASM and instance
```
Comment (dmkozh) [inline src/transactions/test/InvokeHostFunctionTests.cpp:2053]: nit: Wasm

File: src/transactions/test/InvokeHostFunctionTests.cpp [hunk 2]
```diff
@@ -2094,30 +2099,65 @@ TEST_CASE("temp entry eviction", "[tx][soroban]")
...
+
+        REQUIRE(evicted);
+    }
+
+    SECTION("Create temp entry with same key as an expired entry on eviction "
+            "ledger")
     {
-        REQUIRE(lcm.v() == 1);
-        if (lcm.v1().ledgerHeader.header.ledgerSeq == 4097)
+        auto tx2 = test.createInvokeTx(resources, funcSym, {keySym, valU64},
```
Comment (dmkozh) [inline src/transactions/test/InvokeHostFunctionTests.cpp:2140]: IIUC the intention here is for entry to be evicted, will the section above actually be executed before this? (IIRC sections are exclusive, aren't they?)

File: src/transactions/test/InvokeHostFunctionTests.cpp (hunk 1 above)
Comment (sisuresh) [reply src/transactions/test/InvokeHostFunctionTests.cpp]: Fixed

File: src/transactions/test/InvokeHostFunctionTests.cpp (hunk 2 above)
Comment (sisuresh) [reply src/transactions/test/InvokeHostFunctionTests.cpp:2142]: I'm not sure I understand what you mean. The point of this section is to close the ledger where it was supposed to get evicted, but instead in gets recreated.

Comment (dmkozh) [reply src/transactions/test/InvokeHostFunctionTests.cpp:2142]: I see, I thought the intent was to create entry in the ledger following the eviction. This looks fine then.

**Comment by dmkozh** [comment]
r+ 88b2443

**Pull request merged**